src/peerpod-ctrl/Dockerfile

```diff
@@ -1,19 +1,19 @@
 # Build the manager binary
-FROM --platform=$TARGETPLATFORM quay.io/confidential-containers/golang-fedora:1.20.8-38 as builder
+FROM --platform=$TARGETPLATFORM quay.io/confidential-containers/golang-fedora:1.21.9-38 as builder
 ARG TARGETOS
 ARG TARGETARCH
 ARG CGO_ENABLED=1
 ARG GOFLAGS
 
-WORKDIR /workspace/
+WORKDIR /work
```

Comment:
Any specific reason to change the path here?

Reply:
To fix the "plugin was built with a different version of package XXX" from Peerpod-ctrl log.

The go build env must same for `CAA`,` Peerpod-ctrl`, and `the cloud provider plugin`.

Reply:
@liudalibj a basic question. How changing from `/workspace` to `/work` solves the error: `"plugin was built with a different version of package XXX"` ?
Isn't the error tied to the golang runtime ?

Reply:
golang build env MUST same when build external plugin and CAA/peerpod-ctrl binary.
the env include:
- golang verions
- golang pkg path

For this case:
- We build external plugin from `/work`, it make the required `cloud-providers` package hash information will be `/work/cloud-providers/xxx` in the built out *.so
- If we build peerpod-ctrl from `/workspace/`, you can see that the required `cloud-providers` package hash information will be `/workspace/cloud-providers/xxx` in the built out peerpod-ctrl binary.
- So when we try to `plugin.Open` to load the built .so file, go plugin related code will check the `pkghashes`.
https://github.com/golang/go/blob/master/src/runtime/plugin.go#L57

If they are not same then we will see the error `plugin was built with a different version of package XXX`